repo-include.sh

```diff
@@ -0,0 +1,48 @@
+#
+# Copyright (C) 2022  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+# Red Hat Author(s): David Shea <dshea@redhat.com>
+
+TESTTYPE="packaging repo"
+
+. ${KSTESTDIR}/functions.sh
+
+prepare() {
+    local ks=$1
+    local tmp_dir=$2
+    local httpd_url=""
+
+    # Create the repositories 'a' and 'b'.
+    mkdir "${tmp_dir}/http"
+    "${PWD}/scripts/make-addon-pkgs.py" "${tmp_dir}/http"
+
+    # Start a http server to serve the test repo
+    start_httpd "${tmp_dir}/http" "${tmp_dir}"
+
+    # Substitute variables in the kickstart file.
+    sed -e "s|REPO_A_URL|${httpd_url}/a|" \
+        -e "s|REPO_B_URL|${httpd_url}/b|" \
+        "${ks}" > "${tmp_dir}/ks.cfg"
```

Comment:
Is this perhaps a candidate for a helper function? I have seen this at least 4 times now.

Reply:
I wonder if we want that because it's specific for a variable name so you have to have it consistent with the KS file.
But maybe that is not really a problem.